http-scala-fx/src/main/scala/fx/Http.scala

```diff
@@ -0,0 +1,329 @@
+package fx
+
+import java.net.URI
+import java.net.http.HttpHeaders
+import java.net.http.HttpRequest
+import java.net.http.HttpRequest.BodyPublishers
+import java.net.http.HttpResponse
+import java.net.http.HttpResponse.BodyHandlers
+import java.time.Duration
+import java.util.Random
+import scala.annotation.tailrec
+import scala.concurrent.duration.{Duration => ScalaDuration}
+
+/**
+ * Models the http scheme
+ */
+type Http[A] = (
+    HttpRetryPolicy[A],
+    Control[HttpExecutionException],
+    HttpClient
+) ?=> A
+
+extension [A](http: Http[A])
+  def httpValue: Control[HttpExecutionException] ?=> A = http
+  def fmap[B](f: A => B): Http[B] =
+    f(http)
+  def bindMap[B](f: A => Http[B]): Http[B] =
+    f(http)
+
+extension (uri: URI)
+  private def addHeadersToRequestBuilder(
+      builder: HttpRequest.Builder,
+      headers: HttpHeader*): HttpRequest.Builder =
+    headers.foldLeft(builder) { (builder: HttpRequest.Builder, header: HttpHeader) =>
+      header.values.foldLeft[HttpRequest.Builder](builder) {
+        (builder: HttpRequest.Builder, headerValue: String) =>
+          builder.header(header.name, headerValue)
+      }
+    }
+
+  def GET[A](): HttpResponseMapper[A] ?=> Http[HttpResponse[A]] =
+    GET(0, None, List.empty: _*)
+
+  def GET[A](headers: HttpHeader*): HttpResponseMapper[A] ?=> Http[HttpResponse[A]] =
+    GET(0, None, headers: _*)
+
+  def GET[A](
+      timeout: ScalaDuration,
+      headers: HttpHeader*): HttpResponseMapper[A] ?=> Http[HttpResponse[A]] =
+    GET(0, Option(timeout), headers: _*)
+
+  @tailrec
+  private def GET[A](
```

Comment:
I think instead of creating a `def` that returns another function, this  can be declared with `using` as:

```scala
private def GET[A](
      retryCount: Int,
      maybeTimeout: Option[ScalaDuration],
      headers: HttpHeader*)(using mapper: HttpResponseMapper[A]) : Http[HttpResponse[A]]
```
I think the context function makes sense in all cases where we use `val`; otherwise, if we use `def` we are double wrapping. Also, I think this takes care of not having to call `summon` implicits by their types because we can name the injections.

Reply:
Can do. I assume this would be a good thing to bring up in a style guide for the repo as well.

Reply:
Resolved in [c220a86](https://github.com/47deg/TBD/pull/37/commits/c220a86dce47c44f8420cde8b77481c6bb6f7315).